tests/test_qubit_device.py

```diff
@@ -326,8 +327,26 @@ class SomeObservable(qml.operation.Observable):
 
         assert results == [0]
 
+    def test_results_no_state(self, mock_qubit_device_extract_stats, monkeypatch):
+        """Tests that the statistics method raises an AttributeError when a State return type is
+        requested when QubitDevice does not have a state attribute"""
+
+        class SomeObservable(qml.operation.Observable):
+            num_params = 0
+            num_wires = 1
+            par_domain = "F"
+            return_type = State
```

Comment:
We're really trying to get away from annotating observables. This is information that is now absorbed into `MeasurementProcess` with the new core design. I'd recommend not using it here

Reply:
Ok, I did this: 4f2a052ba5553381d000fc8bb45ec5f141a22168